src/gfx_device/gfx_device.rs

```diff
@@ -79,35 +80,50 @@ impl GfxDevice {
                 let mut scene = renderer::Scene::<gfx_device_gl::Resources>::new(camera);
                 let renderables = world.read::<Renderable>();
                 let global_transforms = world.read::<Transform>();
+                // Add all entities which only have Renderable component attached to them to the scene.
+                for (renderable, _) in (&renderables, !&global_transforms).iter() {
+                    // If Transform is not specified use the identity transform.
+                    if let Some(fragment) = unwrap_renderable(renderable, &Transform::default()) {
+                        scene.fragments.push(fragment);
+                    }
+                }
+                // Add all entities which have a Renderable, Transform pair attached to them to the scene.
                 for (renderable, global_transform) in (&renderables, &global_transforms).iter() {
```

Comment:
I think it would be cleaner to just iterate over entities and `renderables`, trying to access the relevant global transform for each entity.